packages/map-template/src/components/MapTemplate/MapTemplate.jsx

```diff
@@ -88,6 +86,133 @@ function MapTemplate({ apiKey, gmApiKey, mapboxAccessToken, venue, locationId, p
 
     const [pushAppView, goBack, currentAppView, currentAppViewPayload, appStates] = useAppHistory();
 
+    // Indicate if the MapsIndoors JavaScript SDK is available
+    const [mapsindoorsSDKAvailable, setMapsindoorsSDKAvailable] = useState(false);
+
+    /**
+     * Ensure that MapsIndoors Web SDK is available.
+     *
+     * @returns {Promise}
+     */
+    function initializeMapsIndoorsSDK() {
+        return new Promise((resolve) => {
+            if (window.mapsindoors !== undefined) {
+                setMapsindoorsSDKAvailable(true);
+                return resolve();
+            }
+
+            const miSdkApiTag = document.createElement('script');
+            miSdkApiTag.setAttribute('type', 'text/javascript');
+            miSdkApiTag.setAttribute('src', 'https://app.mapsindoors.com/mapsindoors/js/sdk/4.23.1/mapsindoors-4.23.1.js.gz');
+            document.body.appendChild(miSdkApiTag);
+            miSdkApiTag.onload = () => {
+                setMapsindoorsSDKAvailable(true);
+                resolve();
+            }
+        });
+    }
+
+    /**
+     * Wait for the MapsIndoors JS SDK to be initialized, then react on changes on the props.
+     * React on changes in the MapsIndoors API key by fetching the required data.
+     */
+    useEffect(() => {
+        initializeMapsIndoorsSDK().then(() => {
```

Comment:
Could the call be done in a useEffect with [] as dependency? I feel like that would be more clean: Why should exactly this useEffect for the apiKey be responsible for initializing the SDK?

Reply:
Hm, just to clarify that I understood correctly. Should we have a separate useEffect for the `initializeMapsIndoorsSDK` and then use the `mapsindoorsSDKAvailable` as a dependency on the useEffect for the apiKey?

Reply:
Sorry for being unclear. But yes 😄

Reply:
I have refactored the `useEffect` - is that the solution that you had in mind?

Reply:
Yup 👍